src/base/events.js

```diff
@@ -165,8 +165,22 @@ export default class Events {
     }
     return this
   }
-}
 
+  static register(eventName) {
+    let property = typeof eventName === 'string' && eventName.toUpperCase().trim()
+
+    if(property && !Events[property]) {
+      Events[property] = 'register:' + property.toLowerCase().replace(/_+/g, '.')
```

Comment:
I don't think custom event values should be prefixed with `register:`. Instead, we could use a camelized version of the event name.

For example:
WILL_CHANGE_SOURCE = 'willChangeSource'

You could do something like this:
```
"WILL_CHANGE_SOURCE".toLowerCase().split('_').map(function(value, index) { return index == 0 ? value : value = (value[0].toUpperCase() + value.slice(1)) }).join('')
```

Another suggestion is to put custom events under a Custom namespace, perhaps something like `Events.Custom.WILL_CHANGE_SOURCE`